Eixozmoviment2/Eixozmovimento2/Eixozmovimento2.pde

```diff
@@ -1,64 +1,76 @@
 float x, y, z;
+float screenW = 600;
+int screenH = 600;
+float a = 0.0;
+float degree = TWO_PI/180.0;
+float cubesPerLevel = 5;
+float angle = 360/5.0;
+int radius = 200;
+
+class Cube { 
+  float ypos, xpos, zpos, speed; 
+  int sizeW, sizeH, cubeLevel;
+  Cube (float x, float y, int w, int h, float s) {  
+    ypos = y; 
+    xpos = x; 
+    zpos = -100;
+    speed = s; 
+    sizeW = w;
+    sizeH = h;
+  } 
+  void level(int cLevel){
+    cubeLevel = cLevel;
+  }
+  void update(float index) { 
+    zpos += speed; 
+    if (ypos > width) { 
+      ypos = 0; 
+    } 
+    float newAngle = angle+(cubeLevel*45);
+    float xPos = xpos+(radius*cos((newAngle*index)*degree));
+    float yPos = ypos+(radius*sin((newAngle*index)*degree));
+    float zPos = zpos-(cubeLevel*100);
+    if(zPos>100){
+      zPos -= 2000;
+    }
+    pushMatrix();
+    translate (xPos, yPos, zPos);
+    fill (255);
+    box (50);
```

Comment:
para alterares o tamanho da box, podes user
box(sizeW, sizeH, 50);

ve a referencia https://processing.org/reference/box_.html

Reply:
"if(zPos>100){
      zPos -= 2000;" o que esta condição significa? Se a posição do array de cubos ja tiver passado de z=100 aparece outro array em z= 2000? eh isto?

E pushMatrix(); ? Não conheco esta propriedade

Reply:
`zPos`, é a variavel que defini para a posição em `z` para cada cubo.
Se a condição `zPos>100` ou seja se a posicao de cada cubo for maior que 100 ja esta para ca do campo de visão da camera imaginaria.

entao se ja esta fora do nosso campo de visao, podemos mover o cubo para uma posicao distante, que faz parecer que esta aparecer um novo cubo, mas estamos so aproveitar os cubos ja existentes.

`pushMatrix` e `popMatrix`, basicamente é abrires e fechar um contexto para fazer o translate de cada cubo, isto tem a ver como processing funciona.

Reply:
https://processing.org/reference/pushMatrix_.html